openslides_backend/action/actions/speaker/create.py

```diff
@@ -255,50 +259,52 @@ def _get_max_weight(self, list_of_speakers_id: int, meeting_id: int) -> int | No
+    def _get_no_speech_type_min(
+        self,
+        list_of_speakers_id: int,
+        meeting_id: int,
+        for_speech_state: SpeechState | None = None,
     ) -> int | None:
-        and_content: list[FilterOperator | Or] = [
-            FilterOperator("list_of_speakers_id", "=", list_of_speakers_id),
+        """
+        Interposed question > intervention & intervention answer > point of order
+        """
+        and_content: list[FilterOperator | Or | And] = [
             Or(
-                And(
-                    FilterOperator("speech_state", "!=", SpeechState.INTERVENTION),
-                    FilterOperator(
-                        "speech_state", "!=", SpeechState.INTERVENTION_ANSWER
-                    ),
-                ),
+                FilterOperator("speech_state", "!=", SpeechState.INTERPOSED_QUESTION),
                 FilterOperator("speech_state", "=", None),
-            ),
-            FilterOperator("begin_time", "=", None),
+            )
         ]
-        if also_point_of_order:
+        if for_speech_state in [
+            SpeechState.INTERVENTION,
+            SpeechState.INTERVENTION_ANSWER,
+            None,
+        ]:
+            and_content.append(
+                Or(
+                    And(
+                        FilterOperator("speech_state", "!=", SpeechState.INTERVENTION),
+                        FilterOperator(
+                            "speech_state", "!=", SpeechState.INTERVENTION_ANSWER
+                        ),
+                    ),
+                    FilterOperator("speech_state", "=", None),
+                )
+            )
+        if for_speech_state is None:
             and_content.append(
                 Or(
                     FilterOperator("point_of_order", "=", False),
                     FilterOperator("point_of_order", "=", None),
                 )
             )
-        return self.datastore.min(
-            collection="speaker",
-            filter=And(and_content),
-            field="weight",
-        )
-
-    def _get_no_interposed_question_min(
-        self, list_of_speakers_id: int, meeting_id: int
-    ) -> int | None:
         return self.datastore.min(
             collection="speaker",
             filter=And(
-                FilterOperator("list_of_speakers_id", "=", list_of_speakers_id),
-                Or(
-                    FilterOperator("speech_state", "=", None),
-                    FilterOperator(
-                        "speech_state", "!=", SpeechState.INTERPOSED_QUESTION
-                    ),
-                ),
-                FilterOperator("begin_time", "=", None),
-                FilterOperator("meeting_id", "=", meeting_id),
+                [
+                    FilterOperator("list_of_speakers_id", "=", list_of_speakers_id),
+                    *and_content,
+                    FilterOperator("begin_time", "=", None),
+                ]
```

Comment:
You can do without creating a new list.
```suggestion
                FilterOperator("list_of_speakers_id", "=", list_of_speakers_id),
                *and_content,
                FilterOperator("begin_time", "=", None),
```

Reply:
Done